eva-server/src/test/java/uk/ac/ebi/eva/server/ws/RegionWSServerTest.java

```diff
@@ -18,68 +18,90 @@
-import com.jayway.restassured.path.json.JsonPath;
-import com.jayway.restassured.response.Response;
-import org.junit.BeforeClass;
+import org.junit.Before;
 import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.opencb.biodata.models.feature.Region;
+import org.opencb.datastore.core.QueryResponse;
+import org.opencb.datastore.core.QueryResult;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import uk.ac.ebi.eva.commons.models.metadata.VariantEntity;
+import uk.ac.ebi.eva.lib.repository.VariantEntityRepository;
 
-import java.net.URI;
 import java.net.URISyntaxException;
-import java.util.Collection;
+import java.util.Arrays;
+import java.util.Collections;
 import java.util.List;
-import java.util.Map;
 
-import static com.jayway.restassured.RestAssured.given;
+import static org.hamcrest.collection.IsCollectionWithSize.hasSize;
+import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertFalse;
-import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertNotEquals;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.argThat;
 
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
 public class RegionWSServerTest {
 
-    @BeforeClass
-    public static void startJetty() throws Exception {
-        RestAssured.port = 8080;
-        RestAssured.baseURI = "http://localhost:8080/eva/webservices/rest";
+    @Autowired
+    private TestRestTemplate restTemplate;
+
+    @MockBean
+    private VariantEntityRepository variantEntityRepository;
+
+    @Before
+    public void setUp() throws Exception {
+        VariantEntity variantEntity = new VariantEntity("chr1", 1000, 1005, "reference", "alternate");
+
+        List<Region> oneRegion = (List<Region>) argThat(hasSize(1));
+        given(variantEntityRepository.findByRegionsAndComplexFilters(oneRegion, any(), any(), any()))
+                .willReturn(Collections.singletonList(variantEntity));
+
+        List<Region> twoRegions = (List<Region>) argThat(hasSize(2));
+        given(variantEntityRepository.findByRegionsAndComplexFilters(twoRegions, any(), any(), any()))
+                .willReturn(Arrays.asList(variantEntity, variantEntity));
     }
 
     @Test
     public void testGetVariantsByRegion() throws URISyntaxException {
-        testGetVariantsByRegionHelper("20:60000-62000", 40);
+        testGetVariantsByRegionHelper("20:60000-62000", 1);
```

Comment:
This way we are not testing what is going on in the web service body.

Imagine that a bug turns the chromosome into an empty string: since the mock repository always returns the same output, independently of the input provided, it would be impossible to detect said bug.

What you could do is to define a small set of regions:
* 20:60000-62000
* 20:61500-62500
* Anything else

And configure the repository to return values for the first 2, and nothing for anything else. That way we can also test the response for non-existing regions quite easily.